ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7483,7 +7483,17 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
             if (ctx.getExplainConfig() != null) {
               writeId = 0L; // For explain plan, txn won't be opened and doesn't make sense to allocate write id
             } else {
-              writeId = txnMgr.getTableWriteId(tblDesc.getDatabaseName(), tblDesc.getTableName());
+              String dbName = tblDesc.getDatabaseName();
+              String tableName = tblDesc.getTableName();
+
+              // CreateTableDesc stores table name as db.table. So, need to decode it before allocating
+              // write id.
+              if ((dbName == null) || tableName.contains(".")) {
+                String[] names = Utilities.getDbTableName(tableName);
+                dbName = names[0];
```

Comment:
if table name does not contain '.' ..this may throw array out of bound exception

Reply:
May be I will remove the dbname null check.